### DISCUSSION TITLE
Buggiest unusable software ever.

### DISCUSSION THREAD

**Discussion by DJGM1974** [unknownskl/xbox-xcloud-client#763] [Q&A]
I've never known such a buggy software such as this. Repeatedly, I'm having to either re-login, or delete and re-install the app.  Sometimes it works OK, but most of the time it just does NOT work. All I get are meaningless gibberish errors.

Greenlight 2.0.0b3 on a 2018 Intel based Mac mini running macOS 13.2.1.

**Comment by DJGM1974** [comment]
I should've pointed out the error messages. Regardless of how often I login or out or restart the app, these errors keeps appearing:


A JavaScript error occurred in the main process

Uncaught Exception:
TypeError: Cannot read properties of undefined (reading 'getProvider')
at IpcMainImpl.<anonymous> (/Applications/Greenlight.app/Contents/Resources/app.asar/app/background.js:1:83881)
at IpcMainImpl.emit (node:events:527:28)
at EventEmitter.<anonymous> (node:electron/js2c/browser_init:165:11237)
at EventEmitter.emit (node:events:527:28)


Error in Promise: 
{"state":"Failed","detailedSessionState":0,"errorDetails":{"code":"WNSError","message":"Xccs : ErrorCallingWNS : Send command failed : State WaitingForServerToRegister”}}


I also get the Windows equivalents of these errors on my PC running Windows 10.

**Comment by DJGM1974** [comment]
Anyone got any suggestions on how to fix this buggy mess?

**Comment by unknownskl** [comment]
Hey! The reauthentication popup is needed because the tokens needs to be retrieved after a application restart. We don't save any tokens locally and that is why. I do agree it isn't the best ux wise but it was needed. 

About the errors, yes they are gibberish to most people. The errors helps the developer to understand what is going wrong. 

The first error is a long running issue and after a while (finally) found out what the problem is. It is a timing issue which occurs when the login action is done before the tokens are retrieved. This issue has been identified and should be fixed in the new beta. (Although other issues can pop up as the authentication flow is completely replaced with a way better authentication mechanism)

The second error about waiting for the server to register is coming from the Xbox api. This means that the Xbox is not able to retrieve the streaming start command and is waiting for the Xbox to register with the Xbox api. This is mostly fixed by restarting the console or by trying to start the stream again. Sometimes it will work for the second time but for some people it doesn't. This is not something we can fix as it is managed by MS.

**Comment by DJGM1974** [comment]
I didn't think I'd need to resurrect this thread. I was up until earlier this week using beta 8 which was working fine until I decided to update to beta 14. Well that was a mistake. Now it seems to constantly stuck on "Loading ... We are getting your stream ready..." and it does not make any connection to the console. It's just stuck on the Loading screen, and that's it.

I logged out of my MS account on Greenlight and logged back in, which temporarily resolved the issue, but next time I came to launch the app it was back to being stuck on "Loading ... We are getting your stream ready..." and not actually connecting to the console.


Greenlight 2.0.0b14 on a 2018 Intel based Mac mini running macOS 13.4.x
